src/rpc/net.cpp

```diff
@@ -587,7 +587,7 @@ static RPCHelpMan getnetworkinfo()
                         {
                             {RPCResult::Type::OBJ, "", "",
                             {
-                                {RPCResult::Type::STR, "name", "network (ipv4, ipv6 or onion)"},
+                                {RPCResult::Type::STR, "name", "network (ipv4, ipv6, onion, i2p, cjdns, or not_publicly_routable)"},
```

Comment:
unrelated nit: Could return a list of all networks in netbase and call `Join(...)` here? Would simplify future review if a network is added or the list added to another place.

Reply:
proposed an update, not sure if it's too dynamic

Reply:
done, it works but not sure if it's kosher to iterate over the enum instead of using a static, manually updated array

Reply:
added regression coverage for the generated help

Reply:
(it seems ok to iterate over the enum per the doc in `src/netaddress.h`)
```
 * Keep these sequential starting from 0 and `NET_MAX` as the last entry.
 * We have loops like `for (int i = 0; i < NET_MAX; i++)` that expect to iterate
 * over all enum values and also `GetExtNetwork()` "extends" this enum by
 * introducing standalone constants starting from `NET_MAX`.
 */
enum Network
```